### PR TITLE
reconfigure main branch for local builds and as scheduler for docs workflows

### PR DESCRIPTION
**Pull request by mojavelinux**
- set up placeholder and trigger for Deploy Docs workflow in docs-build branch
- set up placeholder and trigger for Rebuild Search Index workflow in docs-build branch
- remove obsolete Deploy Reference workflow
- upgrade Antora to 3.1 and tune settings
- reconfigure docs build for local build only
- add patch to support using linked worktree as Antora content source
- remove Antora extensions only needed for the production docs build

### CODE REVIEW

**Comment by mojavelinux** [comment]
Once this PR is merged, the docs-build branch will take over the responsibility of deploying the production docs.

**Review by sjohnr — APPROVED** [review]
This looks good to me! Since I believe you're working with @rwinch I'll let him give a final review before merging.

**Comment by mojavelinux** [comment]
Thanks @sjohnr!

**Review by rwinch — APPROVED** [review]

**Pull request merged**
